src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -51,5 +52,43 @@ public static void ThrowInvalidOperationException_SerializerPropertyNameNull(Jso
         {
             throw new InvalidOperationException(SR.Format(SR.SerializerPropertyNameNull, jsonClassInfo.Type.FullName, jsonPropertyInfo.PropertyInfo.Name));
         }
+
+        public static void ThrowJsonSerializationException_DeserializeDataRemaining(long length, long bytesRemaining)
+        {
+            throw new JsonSerializationException(SR.Format(SR.DeserializeDataRemaining, length, bytesRemaining), 0, 0, "");
+        }
+
+        public static void ThrowJsonSerializationException_DeserializeDuplicateKey(string key, in Utf8JsonReader reader, string path)
+        {
+            ThowJsonSerializationException(SR.Format(SR.DeserializeDuplicateKey, key), in reader, path);
+        }
+
+        private static void ThowJsonSerializationException(string message, in Utf8JsonReader reader, string path)
+        {
+            long lineNumber = reader.CurrentState._lineNumber;
+            long bytePositionInLine = reader.CurrentState._bytePositionInLine;
+
+            message += $" Path: {path} | LineNumber: {lineNumber} | BytePositionInLine: {bytePositionInLine}.";
+            throw new JsonSerializationException(message, lineNumber, bytePositionInLine, path);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowSerializationException(JsonReaderException exception, string path)
+        {
+            string message = exception.Message;
+
+            // Insert the "Path" portion before "LineNumber" and "BytePositionInLine".
+            int iPos = message.LastIndexOf(" LineNumber: ", StringComparison.InvariantCulture);
```

Comment:
Would it be cleaner to append after instead of before? What's the benefit of inserting it before?

Reply:
For serialization IMO the Path is by far the most important. Also this makes it consistent with other serialization exceptions that have Path first and which aren't re-thrown from JsonReaderException.

Reply:
Path may be more important, but it is variable length as well (which pushes out line number/byte position). Since those are effectively fixed size strings, maybe it would be cleaner to have them be up-front. I don't have a strong preference and you make a good point about `Path` being most useful. Inspecting and injecting the text in another exception message seems hacky and fragile to me though.

Reply:
It is a bit hacky. However I did add a test to verify that any changes made in the future to the format of the reader exception would be detected.

Another option which I don't like is to use a general-purpose serialization exception with the path and line number\offset but with no specific message. The user would need to inspect the innerException.